packages/did-provider-ion/src/functions.ts

```diff
@@ -48,9 +55,75 @@ export const toIonPrivateKeyJwk = (privateKeyHex: string): JwkEs256k => {
  * @return The JWK
  */
 export const toIonPublicKeyJwk = (publicKeyHex: string): JwkEs256k => {
-  return toJwkEs256k(secp256k1KeyUtils.publicKeyJwkFromPublicKeyHex(publicKeyHex))
+  return toJwkEs256k(publicKeyJwkFromPublicKeyHex(publicKeyHex))
 }
 
+/**
+ * Example
+ * ```js
+ * {
+ *  kty: 'EC',
+ *  crv: 'secp256k1',
+ *  d: 'rhYFsBPF9q3-uZThy7B3c4LDF_8wnozFUAEm5LLC4Zw',
+ *  x: 'dWCvM4fTdeM0KmloF57zxtBPXTOythHPMm1HCLrdd3A',
+ *  y: '36uMVGM7hnw-N6GnjFcihWE3SkrhMLzzLCdPMXPEXlA',
+ *  kid: 'JUvpllMEYUZ2joO59UNui_XYDqxVqiFLLAJ8klWuPBw'
+ * }
+ * ```
+ * See [rfc7638](https://tools.ietf.org/html/rfc7638) for more details on Jwk.
+ */
+export const getKid = (
+  jwk: ISecp256k1PrivateKeyJwk | ISecp256k1PublicKeyJwk
+) => {
+  const copy = { ...jwk } as any;
+  delete copy.d;
+  delete copy.kid;
+  delete copy.alg;
+  const digest = crypto
+    .createHash('sha256')
+    .update(canonicalize(copy)!, 'utf-8')
+    .digest();
+
+  return base64url.encode(Buffer.from(digest));
```

Comment:
```suggestion
  return u8a.toString(digest, 'base64url')
```

We prefer to avoid Buffer since it requires special treatment on web.
Perhaps with this change we also don't need `base64url` dependency anymore?